src/indexer.c

```diff
@@ -0,0 +1,136 @@
+#include <string.h>
+#include <rmutil/alloc.h>
+#include <rmutil/vector.h>
+
+#include "consts.h"
+#include "indexer.h"
+
+int parseLabel(RedisModuleCtx *ctx, RedisModuleString *label, Label *retLabel, const char *separator) {
+    char *token;
+    char *iter_ptr;
+    size_t _s;
+    const char *labelRaw = RedisModule_StringPtrLen(label, &_s);
+    char *labelstr = RedisModule_PoolAlloc(ctx, _s + 1);
```

Comment:
why is this needed? I see from the code that RedisModule_StringPtrLen returns a null terminated string

Reply:
@yaeltzirulnikov strtok_r manipulates the string its recommend to use it on a copied string.